engine/src/main/java/org/hibernate/validator/internal/engine/cascading/ValueExtractors.java

```diff
@@ -50,23 +50,33 @@ public ValueExtractors(Iterable<ValueExtractor<?>> externalExtractors) {
 		tmpValueExtractors.add( new ValueExtractorDescriptor( LegacyIterableValueExtractor.INSTANCE ) );
 		tmpValueExtractors.add( new ValueExtractorDescriptor( IterableValueExtractor.INSTANCE ) );
 
+		tmpValueExtractors.add( new ValueExtractorDescriptor( OptionalValueExtractor.INSTANCE ) );
 		tmpValueExtractors.add( new ValueExtractorDescriptor( ObjectValueExtractor.INSTANCE ) );
 
 		valueExtractors = Collections.unmodifiableList( tmpValueExtractors );
 	}
 
 	public ValueExtractor<?> getCascadedValueExtractor(Class<?> valueType, TypeVariable<?> cascadingTypeParameter) {
-		Map<Class<?>, Map<TypeVariable<?>, TypeVariable<?>>> allBindings = TypeVariableBindings.getTypeVariableBindings( valueType );
+		Map<Class<?>, Map<TypeVariable<?>, TypeVariable<?>>> allBindings = null;
+
+		if ( cascadingTypeParameter != AnnotatedObject.INSTANCE && cascadingTypeParameter != ArrayElement.INSTANCE ) {
+			allBindings = TypeVariableBindings.getTypeVariableBindings( (Class<?>) cascadingTypeParameter.getGenericDeclaration() );
+		}
```

Comment:
AFAICS it's not a case that can happen but I think I would prefer it to be initialized to an empty map in the other case.